event.go

```diff
@@ -235,6 +235,30 @@ func (e *Event) Strs(key string, vals []string) *Event {
 	return e
 }
 
+// Stringer adds the field key with val as a string to the *Event context.
+// Use Stringer if possible, because it creates string value in lazy mode.
+func (e *Event) Stringer(key string, val fmt.Stringer) *Event {
+	if e == nil {
+		return e
+	}
+	if val == nil {
+		e.buf = enc.AppendInterface(enc.AppendKey(e.buf, key), nil)
+		return e
+	}
+	e.buf = enc.AppendString(enc.AppendKey(e.buf, key), val.String())
```

Comment:
Either this implementation should go to an `AppendStringer` or `AppendStringers` should be moved in this file. It is odd to have this feature implemented in two places.

Reply:
I'm not sure what are you talking about. 
You propose to implement `AppendStriner` for encoder, but it will whole copy of `AppendString` . Or I can call `AppendString ` inside the `AppendStriner `

Reply:
My point is that you repeat the `if val == nil then AppendInterface else AppendString` in many places where it should be a `AppendStringer` in the encoders instead.